File: spring/demo/src/main/java/com/example/demo/entities/Ticket.java
package com.example.demo.entities;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonBackReference;

// @JsonIdentityInfo(
//     generator = ObjectIdGenerators.PropertyGenerator.class,
//     property = "id"
// )

/**
 * Ticket
 */
@Entity
@Table(name="ticket")
public class Ticket {

    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;

    @Column(name="number")
    private int number;

    @Column(name="description")
    private String description;

    @Column(name="priority")
    private String priority;

    @Column(name="open_date")
    private String openDate;

    @Column(name="closed_date")
    private String closedDate;

    @Column(name="remark")
    private String remark;

    @JsonBackReference
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="resolution_category_id")
    public ResolutionCategory resolutionCategoryId; // use singular here
    
    public Long getId() {
        return this.id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public int getNumber() {
        return this.number;
    }

    public void setNumber(int number) {
        this.number = number;
    }

    public String getDescription() {
        return this.description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getPriority() {
        return this.priority;
    }

    public void setPriority(String priority) {
        this.priority = priority;
    }

    public String getOpenDate() {
        return this.openDate;
    }

    public void setOpenDate(String openDate) {
        this.openDate = openDate;
    }

    public String getClosedDate() {
        return this.closedDate;
    }

    public void setClosedDate(String closedDate) {
        this.closedDate = closedDate;
    }

    public String getRemark() {
        return this.remark;
    }

    public void setRemark(String remark) {
        this.remark = remark;
    }

    public ResolutionCategory getResolutionCategoryId() {
        return this.resolutionCategoryId;
    }

    public void setResolutionCategoryId(ResolutionCategory resolutionCategoryId) {
        this.resolutionCategoryId = resolutionCategoryId;
    }


}

File: typeORM/Ticket-Analysis-Dashboard/src/entity/Ticket.ts
import { Entity, PrimaryGeneratedColumn, Column } from "typeorm";
import { ManyToOne } from "typeorm";
import { ResolutionCategory } from "./ResolutionCategory";
import { JoinColumn } from "typeorm";

@Entity()
export class Ticket {
    
    @PrimaryGeneratedColumn()
    id : number;

    @Column()
    number: number;

    @Column()
    description: string;
    
    @Column()
    priority: string;

    @Column({type: "datetime", name: "open_date"})
    openDate;

    @Column({type: "datetime", name: "closed_date"})
    closedDate;

    @Column({name: "resolution_category_id"})
    resolutionCategoryId: number;

    @Column()
    remark: string;

    @ManyToOne(type => ResolutionCategory, resolutionCategory => resolutionCategory.tickets)
    @JoinColumn({name: "resolution_category_id"})
    resolutionCategory: ResolutionCategory



}